core/database/foxx/api/authz.js

```diff
@@ -55,5 +57,182 @@ module.exports = (function() {
     return false;
   };
 
+
+  obj.readRecord = function(client, path) {
+
+    const permission = g_lib.PERM_RD_DATA;
+    // Will split a posix path into an array
+    // E.g.
+    // Will split a posix path into an array
+    // E.g.
+    // path = "/usr/local/bin"
+    // const path_components = pathModule.splitPOSIXPath(path);
+    //
+    // Path components will be
+    // ["usr", "local", "bin"]
+    const path_components = pathModule.splitPOSIXPath(path);
+    const data_key = path_components.at(-1);
+    let record = new Record(data_key);
+
+    if (!record.exists()) {
```

Comment:
I cannot do this for the code I did not write at this point, because there is too much inconsistency, but I will do it for the parts of the code included in the PR.